lib/ansible/modules/windows/win_auto_logon.ps1

```diff
@@ -9,75 +9,75 @@
 
 Set-StrictMode -Version 2.0
 
-$ErrorActionPreferrence = 'Stop'
-
-# argument spec hastable which has the input parameters and capabilities for this module
 $spec = @{
     options = @{
-        domain = @{type = "str"; default = $env:userdomain}
-        user = @{type = "str"; required = $true}
-        password = @{type = "str"; required = $true}
+        password = @{type = "str"; no_log = $true}
         state = @{type = "str"; choices = "absent","present"; default = "present"}
+        username = @{type = "str"}
     }
     required_if = @(
-        @("state", "present", @("user","password")),
-        @("state","absent",@())
+        , @("state", "present", @("username", "password"))
     )
 }
 
-$module = [Ansible.Basic.AnsibleModule]::Create($args,$spec)
-$domain = $module.params.domain
-$user = $module.params.user
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 $password = $module.params.password
 $state = $module.params.state
+$username = $module.params.username
+$domain = $null
 
-try {
-    #Build ParamHash
-
-    $autoAdminLogon = 1
-    $stateMessage = 'added'
-    if($state -eq 'absent'){
-        $autoadminlogon = 0
-        $stateMessage = 'removed'
-    }
-    $module.Result.Msg     = "Auto logon registry keys are already $stateMessage"
-    $autoLogonKeyList   = @{
-        DefaultPassword = $password
-        DefaultUserName = $user
-        DefaultDomain   = $domain
-        AutoAdminLogon  = $autoAdminLogon
+if ($username) {
+    # Try and get the Netlogon form of the username specified. Translating to and from a SID gives us an NTAccount
+    # in the Netlogon form that we desire.
+    $ntAccount = New-Object -TypeName System.Security.Principal.NTAccount -ArgumentList $username
+    try {
+        $accountSid = $ntAccount.Translate([System.Security.Principal.SecurityIdentifier])
+    } catch [System.Security.Principal.IdentityNotMappedException] {
+        $module.FailJson("Failed to find a local or domain user with the name '$username'", $_)
```

Comment:
With this method we can correctly translate an account name to the Netlogon form. As well as this we also can give the user an error if the account isn't valid or resolvable for that host which would fail later on if it tried to auto logon with it.